com.unity.sg2/Editor/GraphUI/EditorCommon/Preview/PreviewUpdateDispatcher.cs

```diff
@@ -0,0 +1,168 @@
+                m_PreviewHandlerInstance.RequestPreviewUpdate(downstreamNode, m_Scheduler);
+        }
+
+        public void OnGlobalPropertyChanged(string propertyName, object newValue)
+        {
+            var linkedVariableNodes =  m_GraphModel.GetLinkedVariableNodes(propertyName);
+
+            var variableNodeNames = new List<string>();
+            foreach(var node in linkedVariableNodes)
+            {
+                var nodeModel = node as GraphDataVariableNodeModel;
+                variableNodeNames.Add(nodeModel.graphDataName);
+            }
+
+            var impactedNodes = m_PreviewHandlerInstance.SetGlobalProperty(propertyName, newValue, variableNodeNames);
+            foreach (var downstreamNode in impactedNodes)
+            {
+                m_PreviewHandlerInstance.RequestPreviewUpdate(downstreamNode, m_Scheduler);
+            }
+        }
+
+        public void OnLocalPropertyChanged(string nodeName, string propertyName, object newValue)
+        {
+            var impactedNodes = m_PreviewHandlerInstance.SetLocalProperty(nodeName, propertyName, newValue);
+            foreach (var downstreamNode in impactedNodes)
+            {
+                m_PreviewHandlerInstance.RequestPreviewUpdate(downstreamNode, m_Scheduler);
+            }
+        }
+
+        bool TimedNodesShouldUpdate(EditorWindow editorWindow)
+        {
+            // get current screen FPS, clamp to what we consider a valid range
+            // this is probably not accurate for multi-monitor.. but should be relevant to at least one of the monitors
+            double monitorFPS = Screen.currentResolution.refreshRate + 1.0;  // +1 to round up, since it is an integer and rounded down
+            if (Double.IsInfinity(monitorFPS) || Double.IsNaN(monitorFPS))
+                monitorFPS = 60.0f;
+            monitorFPS = Math.Min(monitorFPS, 144.0);
+            monitorFPS = Math.Max(monitorFPS, 30.0);
+
+            var curTime = EditorApplication.timeSinceStartup;
+            var deltaTime = curTime - m_LastTimedUpdateTime;
+            bool isFocusedWindow = (EditorWindow.focusedWindow == editorWindow);
+
+            // we throttle the update rate, based on whether the window is focused and if unity is active
+            const double k_AnimatedFPS_WhenNotFocused = 10.0;
+            const double k_AnimatedFPS_WhenInactive = 2.0;
+            double maxAnimatedFPS =
+                (UnityEditorInternal.InternalEditorUtility.isApplicationActive ?
+                    (isFocusedWindow ? monitorFPS : k_AnimatedFPS_WhenNotFocused) :
+                    k_AnimatedFPS_WhenInactive);
+
+            bool update = (deltaTime > (1.0 / maxAnimatedFPS));
+            if (update)
+                m_LastTimedUpdateTime = curTime;
+            return update;
+        }
+
+        public void Update()
+        {
+            if (m_OwningWindowReference == null)
+                return;
+
+            if (TimedNodesShouldUpdate(m_OwningWindowReference))
```

Comment:
TODO: Do this in the PreviewService side of things instead, where we just need to update time in MPB and service can figure out halo and who to update